src/traces/violin/hover.js

```diff
@@ -75,7 +78,15 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode, opts) {
 
                 closeData.push(kdePointData);
 
-                violinLineAttrs = {stroke: pointData.color};
+                var strokeC = pointData.color;
+                var strokeColor = tinycolor(strokeC);
+                var strokeAlpha = strokeColor.getAlpha();
+                var strokeRGB = Color.tinyRGB(strokeColor);
+
+                violinLineAttrs = {
+                    stroke: strokeRGB,
+                    'stroke-opacity': strokeAlpha
+                };
```

Comment:
This one would take a little more work to convert to `Color.stroke` - basically stashing `strokeC` so it can be used below next to the call
```js
violinLine.attr(violinLineAttrs);
```
But I still think it'd be a good idea, and again there's the attr/style switch. In principle both work but we should really be consistent, as there can be different behavior in relation to extra CSS on the page.